### PR TITLE
Improve error messages

### PR DESCRIPTION
**Pull request by Anatoliy-Litvinenko**
Removed unsafe c style read operations.

### CODE REVIEW

File: input.cpp [hunk 1]
```diff
@@ -12,80 +12,95 @@
 
 // functions that read from input files
 namespace marxan {
+    void to_lower(std::string& str)
```
Comment (adelebai) [inline input.cpp]: probably more suitable to be in utils

File: input.cpp [hunk 2]
```diff
@@ -12,80 +12,95 @@
...
-        int temp, newlink;
-
-        for (int i = 0; (i < numvars && foundit == 0); i++)
-        {
-            if (varlist[i].compare(sVarName) == 0)
-                foundit++;
-        }
-
-        if (!head.empty() && find(head.begin(), head.end(), sVarName) != head.end())
+    vector<string> get_tokens(const std::string& str)
```
Comment (adelebai) [inline input.cpp]: in marzone I removed the entire storeFieldName function as well and replaced with this: 

https://github.com/adelebai/marzone/blob/AlgorithmImprovements/util.hpp

See "getFileHeaders" function. I see you've rewritten a similar thing. Is there logic that we could consolidate?

File: input.cpp [hunk 3]
```diff
@@ -12,80 +12,95 @@
...
-        return(sVarName);
+        if(!word.empty())
+            tokens.push_back(word);
+        return tokens;
     }
 
-    vector<string> GetFieldNames(string readname, string fname, FILE*& fp, int& ivars, const vector<string>& varList) {
-        char sLine[500];
+
+    vector<string> GetFieldNames(string readname, string fname, ifstream&  fp, const vector<string>& varList)
```
Comment (adelebai) [inline input.cpp]: See comment above, this function too can be deleted and replaced with a much simpler function. I believe its intention was to just read a line and return the list of header names

File: input.cpp (hunk 2 above)
Comment (adelebai) [reply input.cpp]: Sorry I commented in the wrong spot. Get_tokens looks reasonable and I see you've already removed storeFieldName. What I wanted to comment was actually removing GetFieldNames

File: marxan.cpp [hunk 4]
```diff
@@ -1709,7 +1704,7 @@ namespace marxan {
 
     // determines if the change value for changing a single planning unit status is good
     // does the change stochastically fall below the current acceptance probability?
-    int isGoodChange(const scost& change, double temp, const uniform_real_distribution<double>& float_range)
+    int isGoodChange(const scost& change, double temp, uniform_real_distribution<double>& float_range)
```
Comment (adelebai) [inline marxan.cpp]: I thought I already removed this. Weird

File: input.cpp (hunk 2 above)
Comment (adelebai) [inline input.cpp]: Could you add some unit tests to this? Also I think we should move it to utils since it's more of a general function.

**Review by adelebai — CHANGES_REQUESTED** [review]

File: marxan.cpp (hunk 4 above)
Comment (adelebai) [reply marxan.cpp]: Have you pulled  the latest commits from //newAlgorithm?

**Review by andrist — APPROVED** [review]

File: input.cpp (hunk 1 above)
Comment (Anatoliy-Litvinenko) [reply input.cpp]: Done

File: input.cpp (hunk 2 above)
Comment (Anatoliy-Litvinenko) [reply input.cpp]: Agree. Moving to utils postponed until utils.cpp added to the project.

Comment (adelebai) [reply input.cpp]: There is already a utils.hpp in the project where similar functions sit

**Pull request merged**